checkasm/meson.build

```diff
@@ -0,0 +1,34 @@
+checkasm_src = files(
+    'checkasm.c',
+    'rasterizer.c',
+    'blend_bitmaps.c',
+    'be_blur.c',
+    'blur.c',
+)
+
+checkasm_src_x86 = files(
+    'x86/checkasm.asm',
+)
+checkasm_src_aarch64 = files(
+    'arm/checkasm_64.S',
+)
+
+checkasm_asm_sources = []
+if generic_cpu_family == 'x86'
+    checkasm_src += checkasm_src_x86
+elif generic_cpu_family == 'aarch64'
+    checkasm_src += checkasm_src_aarch64
+endif
```

Comment:
Unlike automake, meson does **not** properly separate compile flags for assembly and C cources (see also `c_args: asm_args` below. That’s why for the main lib non-NASM assembly is split off into an internal static lib, built with `asm_args` while C sources use the project default `c_args`.

As it stands, *all* checkasm sources, even on NASM platforms, do not use the usual C compiler arguments at all iiuc, but only the much smaller and different set meant for non-NASM assembly files (potentially subtly breaking it in some configs but in any event disabling our warnings etc)*

Reply:
Does MSVC accept `-D` macro defines? If not this might actually already break for MSVC Windows ARM builds

Reply:
MSVC does accept `-D` args.

As far as I can tell, the full set of C args *is* passed here; an example command from a local build:
`[40/49] cc -Icheckasm/checkasm.p -Icheckasm -I../checkasm -I. -I.. -Ilibass -I../libass -I/opt/homebrew/opt/freetype/include/freetype2 -I/opt/homebrew/opt/libpng/include/libpng16 -I/opt/homebrew/Cellar/fribidi/1.0.15/include/fribidi -I/opt/homebrew/Cellar/harfbuzz/9.0.0/include/harfbuzz -I/opt/homebrew/Cellar/glib/2.80.4/include/glib-2.0 -I/opt/homebrew/Cellar/glib/2.80.4/lib/glib-2.0/include -I/opt/homebrew/opt/gettext/include -I/opt/homebrew/Cellar/pcre2/10.44/include -I/opt/homebrew/Cellar/graphite2/1.3.14/include -I/opt/homebrew/Cellar/libunibreak/6.1/include -I/opt/homebrew/Cellar/fontconfig/2.14.2/include -fdiagnostics-color=always -Wall -Winvalid-pch -Wextra -std=gnu11 -O2 -g -Wno-sign-compare -Wno-unused-parameter -Werror-implicit-function-declaration -Wstrict-prototypes -Wpointer-arith -Wredundant-decls -Wno-missing-field-initializers -Wformat-non-iso -Werror=format-non-iso -Watomic-implicit-seq-cst -D_GNU_SOURCE -D_XPLATFORM_SOURCE -fno-math-errno -DPREFIX -MD -MQ checkasm/checkasm.p/checkasm.c.o -MF checkasm/checkasm.p/checkasm.c.o.d -o checkasm/checkasm.p/checkasm.c.o -c ../checkasm/checkasm.c`

So it gets everything we passed to `add_project_arguments`, plus `asm_args` (which is just `-DPREFIX`, which is harmless to apply to our C sources as well).

That said, we should really add a ARM builds to the CI (both on Windows and Linux).